pkg/registry/generic/etcd/etcd.go

```diff
@@ -459,16 +462,56 @@ func (e *Etcd) DeleteCollection(ctx api.Context, options *api.DeleteOptions, lis
 	if err != nil {
 		return nil, err
 	}
-	for _, item := range items {
-		accessor, err := meta.Accessor(item)
-		if err != nil {
-			return nil, err
-		}
-		if _, err := e.Delete(ctx, accessor.GetName(), options); err != nil {
-			return nil, err
+	// Spawn a number of goroutines, so that we can issue requests to etcd
+	// in parallel to speed up deletion.
+	workersNumber := registryoptions.GetDeleteCollectionParallelism()
```

Comment:
I've just sent out #21666 out for review to fix this.
I will update this PR once that one is reviewed.